### PR TITLE
Some method-to-ir cleanup.

### PR DESCRIPTION
**Pull request by jaykrell**

### CODE REVIEW

File: mono/mini/method-to-ir.c [hunk 1]
```diff
@@ -8592,72 +8609,26 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 				}
 			}
 			ins = (MonoInst*)mini_emit_calli_full (cfg, fsig, sp, addr, NULL, NULL, tailcall);
-
```
Comment (vargaz) [inline mono/mini/method-to-ir.c]: I don't think we should goto between switch cases, its really confusing. Just keep the duplicate code.

Comment (jaykrell) [reply mono/mini/method-to-ir.c]: I think it is too much duplicated code. It was less before, and it has incorrectly diverged even then. Now it is larger, though maybe could be dialed back.

Ultimately I think the common tail becomes a function and then can call it from two places.

File: mono/mini/method-to-ir.c [hunk 2]
```diff
@@ -8583,87 +8598,38 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
...
 				} else if (info_type == MONO_PATCH_INFO_JIT_ICALL_ADDR) {
-					// FIXME tailcall?
-					tailcall_calli = FALSE;
+					tailcall = FALSE;
 					ins = (MonoInst*)mono_emit_abs_call (cfg, info_type, info_data, fsig, sp);
 					NULLIFY_INS (addr);
 					goto calli_end;
 				}
 			}
-			ins = (MonoInst*)mini_emit_calli_full (cfg, fsig, sp, addr, NULL, NULL, tailcall_calli);
```
Comment (jaykrell) [inline mono/mini/method-to-ir.c]: A whole screenful of duplication (for me), and it was missing the sequence point logic.

**Comment by jaykrell** [comment]
@monojenkins build Linux x64

**Comment by jaykrell** [comment]
@monojenkins build OS X x64

File: mono/mini/method-to-ir.c (hunk 2 above)
Comment (jaykrell) [reply mono/mini/method-to-ir.c]: I don't want this duplication. I can try writing a function for it and call it from two places.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build Linux i386

**Comment by jaykrell** [comment]
@monojenkins build Linux AArch64 Interpreter

**Pull request merged**